Objects/exceptions.c

```diff
@@ -2994,52 +2994,53 @@ UnicodeEncodeError_init(PyObject *self, PyObject *args, PyObject *kwds)
 static PyObject *
 UnicodeEncodeError_str(PyObject *self)
 {
-    PyUnicodeErrorObject *uself = (PyUnicodeErrorObject *)self;
-    PyObject *result = NULL;
-    PyObject *reason_str = NULL;
-    PyObject *encoding_str = NULL;
```

Comment:
Is there a reason you change the error handling from `goto done;` to doing `Py_DECREF` & `return` throughout the function?
I find the former much easier to maintain, since a new owned reference/resource only needs to be added to the beginning and end.

Reply:
I would usually keep the goto if we had lots of paths but here we only have two real jumps and only one would actually need a 'goto' (for `encoding`) so I though it could be cleaner.

But I can revert it if you want a smaller diff.

Reply:
I've reverted the changes though I would have preferred to keep them. The diff is definitely nicer but I think the code looks a bit uglier :smile:

Reply:
Well, adding braces around `if` bodies was a good change, despite the diff. Do you still want to add those?

Reply:
Mmh. I'll add braces. At least, if we want one day to change it to what I wrote, braces would already be there :')